titiler/cmr/backend.py

```diff
@@ -172,14 +174,30 @@ def get_assets(
 
         assets: List[Asset] = []
         for r in results:
-            assets.append(
-                {
-                    "url": r.data_links(access="direct")[
-                        0
-                    ],  # NOTE: should we not do this?
-                    "provider": r["meta"]["provider-id"],
-                }
-            )
+            if bands_regex:
```

Comment:
@vincentsarago This looks good to me.  In the future we may want consider a more explicit mapping of band id->band regex. Something like `bands_regex=.*B10\.tif.*&bands_regex=.*B09\.tif.*` so that we are only linking the same assets listed in the bands parameter but I think the approach you have here should be good until we find a dataset that breaks it :]